### DISCUSSION TITLE
A simple prefix on FormAction

### DISCUSSION THREAD

**Discussion by yixe** [pointfreeco/swift-composable-architecture#565] [Q&A]
is there a simpler way to achieve a prefix on a simple form field?

```swift
viewStore.binding(keyPath: \.username, send: Profile.form)
```

i'd link to add a simple .prefix(10) to this field

**Comment by tgrapperon** [comment]
If I understand correctly, you can use a computed property if you want to continue using a `binding` reducer:
```swift
var userName10: String {
  get { username.prefix(10) }
  set { username = newValue.prefix(10) }
}
```
and use `\.username10` in your binding.
I would personally implement this validation logic in a reducer though, but it's mostly a matter of taste and it depends on the complexity of your domain. I guess doing the validation in the accessor is OK in this simple case.
